src/py/mat3ra/made/tools/calculate.py

```diff
@@ -167,3 +167,100 @@ def calculate_norm_of_distances(material: Material, shadowing_radius: float = 2.
     substrate_coordinates_values = np.array(substrate_atoms_surface_coordinates.values)
 
     return get_norm_of_distances_between_coordinates(film_coordinates_values, substrate_coordinates_values)
+
+
+class SurfaceDistanceCalculator(ASECalculator):
```

Comment:
FixedZ

Reply:
Uhm, we pass this as a paramter: whether to fix_z, fix_substrate or not

Reply:
FilmSubstrateDistanceCalculator or InterfacialSeparationDistanceCalculator